From a8270098e9d18044e9e8237e916ac093fad5a9b5 Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Tue, 4 Oct 2022 10:27:14 +0200
Subject: [PATCH 3/3] ramips: disable switch/bridge offload

This is breaking win10 laptops when vlan_filtering is enabled

Signed-off-by: John Crispin <john@phrozen.org>
---
 .../linux/ramips/patches-5.10/992-dsa.patch   | 33 +++++++++++++++++++
 1 file changed, 33 insertions(+)
 create mode 100644 target/linux/ramips/patches-5.10/992-dsa.patch

diff --git a/target/linux/ramips/patches-5.10/992-dsa.patch b/target/linux/ramips/patches-5.10/992-dsa.patch
new file mode 100644
index 0000000000..67b214e5a9
--- /dev/null
+++ b/target/linux/ramips/patches-5.10/992-dsa.patch
@@ -0,0 +1,33 @@
+Index: linux-5.10.138/drivers/net/dsa/mt7530.c
+===================================================================
+--- linux-5.10.138.orig/drivers/net/dsa/mt7530.c
++++ linux-5.10.138/drivers/net/dsa/mt7530.c
+@@ -3018,20 +3018,20 @@ static const struct dsa_switch_ops mt753
+ 	.port_max_mtu		= mt7530_port_max_mtu,
+ 	.port_stp_state_set	= mt7530_stp_state_set,
+ 	.port_egress_floods	= mt7530_port_egress_floods,
+-	.port_bridge_join	= mt7530_port_bridge_join,
+-	.port_bridge_leave	= mt7530_port_bridge_leave,
++//	.port_bridge_join	= mt7530_port_bridge_join,
++//	.port_bridge_leave	= mt7530_port_bridge_leave,
+ 	.port_fdb_add		= mt7530_port_fdb_add,
+ 	.port_fdb_del		= mt7530_port_fdb_del,
+ 	.port_fdb_dump		= mt7530_port_fdb_dump,
+ 	.port_mdb_prepare	= mt7530_port_mdb_prepare,
+ 	.port_mdb_add		= mt7530_port_mdb_add,
+ 	.port_mdb_del		= mt7530_port_mdb_del,
+-	.port_vlan_filtering	= mt7530_port_vlan_filtering,
+-	.port_vlan_prepare	= mt7530_port_vlan_prepare,
+-	.port_vlan_add		= mt7530_port_vlan_add,
+-	.port_vlan_del		= mt7530_port_vlan_del,
+-	.port_mirror_add	= mt753x_port_mirror_add,
+-	.port_mirror_del	= mt753x_port_mirror_del,
++//	.port_vlan_filtering	= mt7530_port_vlan_filtering,
++//	.port_vlan_prepare	= mt7530_port_vlan_prepare,
++//	.port_vlan_add		= mt7530_port_vlan_add,
++//	.port_vlan_del		= mt7530_port_vlan_del,
++//	.port_mirror_add	= mt753x_port_mirror_add,
++//	.port_mirror_del	= mt753x_port_mirror_del,
+ 	.phylink_validate	= mt753x_phylink_validate,
+ 	.phylink_mac_link_state	= mt753x_phylink_mac_link_state,
+ 	.phylink_mac_config	= mt753x_phylink_mac_config,
-- 
2.25.1

